ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java

```diff
@@ -108,21 +116,30 @@ private ClientImpl(final ClientOptions clientOptions, final Vertx vertx,
     this.serverSocketAddress =
         SocketAddress.inetSocketAddress(clientOptions.getPort(), clientOptions.getHost());
     this.sessionVariables = new HashMap<>();
+    this.serializedConsistencyVector = new AtomicReference<>();
+    this.requestProperties = new HashMap<>();
+    this.parser = new LocalPropertyParser();
   }
 
   @Override
   public CompletableFuture<StreamedQueryResult> streamQuery(final String sql) {
-    return streamQuery(sql, Collections.emptyMap());
+    return streamQuery(sql, new HashMap<>());
   }
 
   @Override
   public CompletableFuture<StreamedQueryResult> streamQuery(
       final String sql,
       final Map<String, Object> properties
   ) {
+    if (ConsistencyOffsetVector.isConsistencyVectorEnabled(requestProperties)) {
+      requestProperties.put(
+          KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR,
+          serializedConsistencyVector.get());
```

Comment:
Can this be null rather than empty string if it's been unset?

Reply:
Yeah, I had it as null before but the code for `JsonObject.put` throws an NPE